.github/workflows/e2e-direct-commit-pr-test.yml

```
@@ -0,0 +1,431 @@
+            echo "🎯 Cypress dependencies installation completed"
+
+          elif [[ "$RUNNER_OS" == "macOS" ]]; then
+            echo "🍎 macOS detected - installing Xvfb..."
+            brew install xorg-server || echo "Homebrew install failed, Xvfb may already be available"
+
+          elif [[ "$RUNNER_OS" == "Windows" ]]; then
+            echo "🪟 Windows detected - Cypress dependencies usually pre-installed"
+          fi
+
+      - name: Debug available secrets
+        run: |
```

actionlint (1.7.7)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)